conda_smithy/lint_recipe.py

```diff
@@ -679,13 +679,28 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
     if is_staged_recipes and recipe_name:
         cf = gh.get_user(os.getenv("GH_ORG", "conda-forge"))
         try:
-            cf.get_repo("{}-feedstock".format(recipe_name))
-            feedstock_exists = True
+            for name in set(
+                [
+                    recipe_name,
+                    recipe_name.replace("-", "_"),
+                    recipe_name.replace("_", "-"),
+                ]
+            ):
+                if cf.get_repo("{}-feedstock".format(name)):
+                    existing_recipe_name = name
+                    feedstock_exists = True
+                    break
+            else:
+                feedstock_exists = False
         except github.UnknownObjectException as e:
             feedstock_exists = False
 
         if feedstock_exists:
-            lints.append("Feedstock with the same name exists in conda-forge")
+            lints.append(
+                'Feedstock with the name "{}" exists in conda-forge'.format(
+                    existing_recipe_name
```

Comment:
This is better, but can we be more specific. If the name doesn't match, the message should say something to the effect of "We found a feedstock with the name {}. It doesn't match yours but we suspect it may be the same package." In this case it should be a linting hint.

If the name matches, we can keep the current message and keep it as a lint.